zulipterminal/ui_tools/boxes.py

```diff
@@ -192,11 +203,14 @@ def stream_header(self) -> Any:
         return header
 
     def private_header(self) -> Any:
-        self.recipients = ', '.join(list(
-            recipient['full_name']
-            for recipient in self.message['display_recipient']
-            if recipient['email'] != self.model.user_email
-        ))
+        if self._is_private_message_to_self():
+            self.recipients = self.message['display_recipient'][0]['full_name']
+        else:
+            self.recipients = ', '.join(list(
+                recipient['full_name']
+                for recipient in self.message['display_recipient']
+                if recipient['email'] != self.model.user_email
+            ))
```

Comment:
As per my comment on the previous commit, we don't now use `self.recipients` :)

Reply:
Are both the comments you left here referring to the same thing? Which is I need to change in `private_header` function as I am still referring to the old `self.recipients` variable there.
Re tests: Will add them :)

Apologies about this. Missed to spot this. :(

Reply:
Yes, same area - I think we clarified this on czo?